src/backend/marsha/core/xapi.py

```diff
@@ -83,28 +102,32 @@ class XAPIDocumentStatement(XAPIStatementMixin):
     """Object managing statement for document objects."""
 
     # pylint: disable=too-many-arguments
-    def _build_statement(self, document, statement, homepage, user=None, user_id=None):
+    def _build_statement(
+        self, document, statement, homepage, user=None, user_id=None, username=None
+    ):
         """Build all common properties for a document."""
 
         if re.match(r"^http(s?):\/\/.*", homepage) is None:
             homepage = f"http://{homepage}"
 
         statement = self.build_common_statement_properties(
-            statement, homepage, user=user, user_id=user_id
+            statement, homepage, user=user, user_id=user_id, username=username
         )
 
         statement["context"].update(
-            {"contextActivities": {"category": [{"id": "https://w3id.org/xapi/lms"}]}}
+            {
+                "contextActivities": {
+                    "category": [LMSProfileActivity().model_dump(exclude_none=True)]
+                }
+            }
         )
 
-        statement["object"] = {
-            "definition": {
-                "type": "http://id.tincanapi.com/activitytype/document",
-                "name": {self.get_locale(): document.title},
-            },
-            "id": f"uuid://{document.id}",
-            "objectType": "Activity",
-        }
+        statement["object"] = DocumentActivity(
+            definition=DocumentActivityDefinition(
+                name={self.get_locale(): document.title}
+            ),
+            id=f"uuid://{document.id}",
+        ).model_dump(exclude_none=True)
```

Comment:
Would it be possible to use `BaseXapiStatement` from ralph to build these core properties?